src/components/editor/Bindings/SchemaInference/Dialog/DiffEditor/index.tsx

```diff
@@ -4,33 +4,39 @@ import { BindingsEditorSchemaSkeleton } from 'components/collection/CollectionSk
 import InferenceDiffEditorFooter from 'components/editor/Bindings/SchemaInference/Dialog/DiffEditor/Footer';
 import InferenceDiffEditorHeader from 'components/editor/Bindings/SchemaInference/Dialog/DiffEditor/Header';
 import {
+    useBindingsEditorStore_collectionData,
     useBindingsEditorStore_inferredSpec,
     useBindingsEditorStore_loadingInferredSchema,
 } from 'components/editor/Bindings/Store/hooks';
-import { CollectionData } from 'components/editor/Bindings/types';
 import { DEFAULT_HEIGHT } from 'components/editor/MonacoEditor';
 import AlertBox from 'components/shared/AlertBox';
 import {
     defaultOutline,
     monacoEditorComponentBackground,
     monacoEditorWidgetBackground,
 } from 'context/Theme';
+import { useMemo } from 'react';
 import { FormattedMessage } from 'react-intl';
 import { stringifyJSON } from 'services/stringify';
 
 interface Props {
-    collectionData: CollectionData;
     height?: number;
 }
 
-function InferenceDiffEditor({
-    collectionData,
-    height = DEFAULT_HEIGHT,
-}: Props) {
+function InferenceDiffEditor({ height = DEFAULT_HEIGHT }: Props) {
     const theme = useTheme();
 
     // Bindings Editor Store
     const inferredSpec = useBindingsEditorStore_inferredSpec();
+    const collectionData = useBindingsEditorStore_collectionData();
+
+    const original = useMemo(() => {
+        if (collectionData) {
+            return stringifyJSON(collectionData.spec);
+        }
+
+        return null;
+    }, [collectionData]);
```

Comment:
Now that the `collectionData` is read from the store it is easier to handle this check in a memo.